pkg/apis/v1alpha1/cluster_template.go

```diff
@@ -56,6 +58,8 @@ type TemplateSpec struct {
 	// interpolation using the $()$ marker format. For more
 	// information, see: https://cartographer.sh/docs/latest/templating/
 	// You cannot define both Template and Ytt at the same time.
+	// You cannot define the namespace for the resource. The resource will
+	// be created in the owner namespace.
```

Comment:
This sounds a bit like we'll ignore the namespace and just create in the owner namespace, rather than fail (which is the behavior this PR is introducing). Is there a way we can make that a bit clearer? i.e. something like:
```
You should not define the namespace for the resource- it will automatically be created in the owner namespace. If the namespace is specified and is not the owner namespace, the resource will fail to be created.
```